staging/src/k8s.io/apiserver/pkg/admission/config_test.go

```diff
@@ -192,12 +214,18 @@ func TestReadAdmissionConfiguration(t *testing.T) {
 			t.Fatalf("unexpected err writing temp file: %v", err)
 		}
 		config, err := ReadAdmissionConfiguration(testCase.PluginNames, configFileName, scheme)
-		if err != nil {
+		if err != nil && len(testCase.ExpectedError) == 0 {
 			t.Fatalf("unexpected err: %v", err)
 		}
-		if !reflect.DeepEqual(config.(configProvider).config, testCase.ExpectedAdmissionConfig) {
+		if err == nil && len(testCase.ExpectedError) > 0 {
+			t.Fatalf("expected error but received none")
+		}
+		if err != nil && len(testCase.ExpectedError) > 0 {
+			assert.Contains(t, err.Error(), testCase.ExpectedError)
+		} else if !reflect.DeepEqual(config.(configProvider).config, testCase.ExpectedAdmissionConfig) {
```

Comment:
may be check expectedError first is better ?
```go
if testCase.ExpectedError != "" {
	if err != nil {
		assert.Contains(t, err.Error(), testCase.ExpectedError)
	} else {
		t.Fatalf("expected error %q but received none", testCase.ExpectedError)
	}
} else if err != nil {
	t.Fatalf("unexpected err: %v", err)
}


if !reflect.DeepEqual(config.(configProvider).config, testCase.ExpectedAdmissionConfig) {
	t.Fatalf("unexpected admission config: expected %v, got %v", testCase.ExpectedAdmissionConfig, config.(configProvider).config)
}
```

Reply:
Updated. Please note that the final `if !reflect.DeepEqual...` must be an `else if ...` statement. This final statement should only be run if we do not expect an error and if one has not occurred.

Reply:
ok, thank you very much.